### PR TITLE
Use `LazyArtifacts` package when the artifact is lazy

### PR DESCRIPTION
**Pull request by giordano**
Trying to load `MKL_jll` on Julia v1.6 I get

```
ERROR: LoadError: LoadError: InitError: Artifact "MKL" is a lazy artifact; package developers must call `using LazyArtifacts` in MKL_jll before using lazy artifacts.
```

This PR is meant to address this issue, but I'm not 100% sure this is correct.
I already created
[`LazyArtifacts.jl`](https://github.com/JuliaPackaging/LazyArtifacts.jl) for
compatibility with older Julia versions, but not yet registered it.

### CODE REVIEW

**Comment by staticfloat** [comment]
Yep, this seems right.

File: src/AutoBuild.jl [hunk 1]
```diff
@@ -1345,7 +1350,7 @@ const uuid_package = UUID("cfb74b52-ec16-5bb7-a574-95d9e393895e")
 # For even more interesting historical reasons, we append an extra
 # "_jll" to the name of the new package before computing its UUID.
 jll_uuid(name) = bb_specific_uuid5(uuid_package, "$(name)_jll")
-function build_project_dict(name, version, dependencies::Array{Dependency}, julia_compat::String=DEFAULT_JULIA_VERSION_SPEC)
+function build_project_dict(name, version, dependencies::Array{Dependency}, julia_compat::String=DEFAULT_JULIA_VERSION_SPEC, lazy_artifacts::Bool=false)
```
Comment (staticfloat) [inline src/AutoBuild.jl:1353]: I think the callers of `build_project_dict()` will need to pass `lazy_artifacts` through as well.

Comment (giordano) [reply src/AutoBuild.jl]: Ooops, good point!  I _think_ I've fixed that

**Comment by giordano** [comment]
Are we supposed to only load `LazyArtifacts` or do also something else?  In the `MKL_jll` generated with this branch I get
```
julia> using MKL_jll
  Downloaded artifact: MKL
ERROR: InitError: Unable to automatically install 'MKL' from '/home/mose/.julia/dev/MKL_jll/Artifacts.toml'
Stacktrace:
  [1] error(s::String)
    @ Base ./error.jl:33
  [2] ensure_artifact_installed(name::String, meta::Dict{String, Any}, artifacts_toml::String; platform::Base.BinaryPlatforms.Platform, verbose::Bool, quiet_download::Bool)
    @ Pkg.Artifacts ~/repo/julia/usr/share/julia/stdlib/v1.6/Pkg/src/Artifacts.jl:424
  [3] ensure_artifact_installed(name::String, artifacts_toml::String; platform::Base.BinaryPlatforms.Platform, pkg_uuid::Nothing, verbose::Bool, quiet_download::Bool)
    @ Pkg.Artifacts ~/repo/julia/usr/share/julia/stdlib/v1.6/Pkg/src/Artifacts.jl:388
  [4] _artifact_str(__module__::Module, artifacts_toml::String, name::SubString{String}, path_tail::String, artifact_dict::Dict{String, Any}, hash::Base.SHA1, platform::Base.BinaryPlatforms.Platform, lazyartifacts::Any)
    @ Artifacts ~/repo/julia/usr/share/julia/stdlib/v1.6/Artifacts/src/Artifacts.jl:546
  [5] invokelatest(::Any, ::Any, ::Vararg{Any, N} where N; kwargs::Base.Iterators.Pairs{Union{}, Union{}, Tuple{}, NamedTuple{(), Tuple{}}})
    @ Base ./essentials.jl:707
  [6] invokelatest(::Any, ::Any, ::Vararg{Any, N} where N)
    @ Base ./essentials.jl:706
  [7] macro expansion
    @ ~/repo/julia/usr/share/julia/stdlib/v1.6/Artifacts/src/Artifacts.jl:675 [inlined]
  [8] find_artifact_dir()
    @ MKL_jll ~/.julia/packages/JLLWrappers/HEjmb/src/wrapper_generators.jl:15
  [9] __init__()
    @ MKL_jll ~/.julia/dev/MKL_jll/src/wrappers/x86_64-linux-gnu.jl:9
 [10] _include_from_serialized(path::String, depmods::Vector{Any})
    @ Base ./loading.jl:669
 [11] _require_search_from_serialized(pkg::Base.PkgId, sourcepath::String)
    @ Base ./loading.jl:755
 [12] _require(pkg::Base.PkgId)
    @ Base ./loading.jl:993
 [13] require(uuidkey::Base.PkgId)
    @ Base ./loading.jl:909
 [14] require(into::Module, mod::Symbol)
    @ Base ./loading.jl:896
during initialization of module MKL_jll
```

**Comment by staticfloat** [comment]
That looks to me like some kind of artifact installation error, not a problem with `LazyArtifacts`.

**Comment by giordano** [comment]
Uh uh, the `Artifacts.toml` was in a bad shape, fixing that the package loads fine

**Pull request merged**